### PR TITLE
Use version comparison for autoconf compatibility

### PR DESCRIPTION
**Pull request by emillon**
`major < 5 && minor < 8` fails for 3.12 (and probably for 5.0).
Instead, we use `compare` (through `Has`) which does the right thing.
A small test suite is added to show the corner cases.

### CODE REVIEW

File: ocaml_version.ml [hunk 1]
```diff
@@ -304,16 +311,8 @@ module Configure_options = struct
...
-      match o with
-      | `Afl -> "-afl-instrument"
-      | `Flambda -> "-flambda"
-      | `Default_unsafe_string -> "-default-unsafe-string"
-      | `Force_safe_string -> "-force-safe-string"
-      | `Frame_pointer -> "-with-frame-pointer"
-      | _ -> ""
-    else (* post autoconf *)
+  let to_configure_flag t o =
+    if Has.autoconf t then
```
Comment (kit-ty-kate) [inline ocaml_version.ml:315]: shouldn't it be `not (Has.autoconf t)` instead?

File: ocaml_version.ml [hunk 2]
```diff
@@ -234,6 +236,11 @@ module Has = struct
     match compare (Since.arch a) v with
     |(-1) | 0 -> true
     |_ -> false
+
+  let autoconf v =
+    match compare Since.autoconf v with
+    |(-1) | 0 -> true
```
Comment (kit-ty-kate) [inline ocaml_version.ml:242]: souldn't `Has.autoconf` too if the version is `> 4.08` instead of `<= 4.08`?

File: ocaml_version.ml (hunk 1 above)
Comment (emillon) [reply ocaml_version.ml:315]: The branches are flipped, compared to `master`. So the "has autoconf" / "post autoconf" branch is the `then` one now.

File: ocaml_version.ml (hunk 2 above)
Comment (emillon) [reply ocaml_version.ml:242]: That's what it does (this is consistent with other functions in `Has`):
`Has.autoconf v`
<=>
`compare Since.autoconf v <= 0`
<=>
`Since.autoconf <= v`
<=>
`v >= 4.08.0`

File: ocaml_version.ml (hunk 1 above)
Comment (kit-ty-kate) [reply ocaml_version.ml:315]: oh, I see, code review on my phone was a bad idea.. I'm so sorry for the disturbance.

Comment (emillon) [reply ocaml_version.ml:315]: no problem I was confused with these comparisons as well (which prompted me to write a couple tests)

**Review by kit-ty-kate — APPROVED** [review]
I'm so sorry for the bad code review I just did. I blame tiredness and the octodroid app (but mainly tiredness I guess :sweat_smile:), sorry again.

Looks really good.
